src/runtime_strings_extra.js

```diff
@@ -39,7 +39,8 @@ var UTF16Decoder = typeof TextDecoder !== 'undefined' ? new TextDecoder('utf-16l
 #endif // TEXTDECODER
 #endif // TEXTDECODER == 2
 
-function UTF16ToString(ptr) {
+function UTF16ToString(ptr, maxCharsToRead) {
+  if (maxCharsToRead === undefined) maxCharsToRead = Infinity;
```

Comment:
I think we should be consistent with how `UTF8ArrayToString` does things, it has that trick,
```javascript
  // (As a tiny code save trick, compare endPtr against endIdx using a negation, so that undefined means Infinity)
```
We should either do them all that way, or none, I think, so it's consistent one way or the other. I lean towards doing the trick, with that documentation everywhere.

Reply:
Makes sense (and I'd missed that comment so I was more befuddled than I needed to be when I first saw it), consistency makes sense.